Recast/iTunesPodcasts/Models/RSS/Enclosure.swift

```diff
@@ -13,31 +13,37 @@ public enum Enclosure {
     // supported file formats are: audio/x-m4a, audio/mpeg, video/quicktime,
     // video/mp4, video/x-m4v, and application/pdf.
     
+    public enum MediaType: String {
+        case m4a = "audio/x-m4a"
+        case mpeg = "audio/mpeg"
+        case quicktime = "video/quicktime"
+        case mp4 = "video/mp4"
+        case m4v = "video/x-m4v"
+        case pdf = "application/pdf"
+    }
+    
     // url is asset url, length is size in bytes
-    case audio(url: URL, length: Int)
-    case video(url: URL, length: Int)
+    case audio(url: URL, length: Int, type: MediaType)
+    case video(url: URL, length: Int, type: MediaType)
     case pdf(url: URL, length: Int)
     
     init?(from attributes: [String : String]) {
         guard !attributes.isEmpty,
-            let type = attributes["type"],
+            let typeStr = attributes["type"],
+            let type = MediaType(rawValue: typeStr),
             let url = URL(string: attributes["url"] ?? ""),
             let length = Int(attributes["length"] ?? "") else {
                 return nil
         }
         
-        let audio_types = ["audio/x-m4a", "audio/mpeg"]
-        let video_types = ["video/quicktime", "video/mp4", "video/x-m4v"]
-        let pdf_types = ["application/pdf"]
-        if audio_types.contains(type)  {
-            self = .audio(url: url, length: length)
-            return
-        } else if video_types.contains(type) {
-            self = .video(url: url, length: length)
-            return
-        } else if pdf_types.contains(type) {
+        
+        switch type {
+        case .m4a, .mpeg:
+            self = .audio(url: url, length: length, type: type)
+        case .quicktime, .mp4, .m4v:
+            self = .video(url: url, length: length, type: type)
+        case .pdf:
             self = .pdf(url: url, length: length)
-            return
         }
         return nil
```

Comment:
this `return nil` will always be executed, you sure this is what we want from our init? or should that be in `case default:`

Reply:
ahh nah I need to insert regular returns